### PR TITLE
Offset: Increase search depth when finding the 'real' offset parent

### PR DESCRIPTION
**Pull request by Minimaximize**
### Summary ###
Continue searching for the next positioned offsetParent (or the document) when calculating `position()`.

This fixes the seemingly inconsistent behaviour that occurred when calling `position()` on elements inside tables; for which the [offsetParent](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent) changes based on its `position` css property. e.g
```
<div id="positioned-container" style="position: relative;">
    <table>
        <tr>
            <td>
                <span id="static"></span>
                <span id="relative" style="position: relative;"></span>
                <span id="absolute" style="position: absolute;"></span>
            </td>
        </tr>
    </table>
</div>

The offsetParent of #static is td
The offsetParent of #relative and #absolute is #positioned-container
```

Previously `$('#static').position()` was returning the position relative to the containing `<td>` element, while `$('#relative').position()` was returning the position relative to `#positioned-container`

Now, both elements return their position relative to the containing `#positioned-container`

### Checklist ###

* [x] All authors have signed the CLA at https://cla.js.foundation/jquery/jquery
* [x] New tests have been added to show the fix or feature works
* [x] Grunt build and unit tests pass locally with these changes
* [ ] If needed, a docs issue/PR was created at https://github.com/jquery/api.jquery.com

### CODE REVIEW

**Comment by Minimaximize** [comment]
Whoops, I thought I had fixed that failing test 😓😓

**Comment by Minimaximize** [comment]
Finally tracked down the docs describing why the licence check was indeterminate - Fixed the erroneous author attributions for my commits

**Comment by mgol** [comment]
Closing & re-opening the PR to trigger the EasyCLA check...

**Review by gibson042 — APPROVED** [review]
Thanks, this is good!

File: src/offset.js [hunk 1]
```diff
@@ -122,12 +122,13 @@ jQuery.fn.extend( {
...
 
-				offsetParent = offsetParent.parentNode;
+				offsetParent = offsetParent.offsetParent || doc.documentElement;
 			}
-			if ( offsetParent && offsetParent !== elem && offsetParent.nodeType === 1 ) {
+			if ( offsetParent && offsetParent !== elem && offsetParent.nodeType === 1 &&
+				jQuery.css( offsetParent, "position" ) !== "static" ) {
 
 				// Incorporate borders into its offset, since they are outside its content origin
 				parentOffset = jQuery( offsetParent ).offset();
```
[the remark this answers is not in the document]
Comment (mgol) [reply src/offset.js:134]: I guess we should add some tests for this part of the behavior then as well.

[the remark this answers is not in the document]
Comment (mgol) [reply src/offset.js:134]: Extracted to #5468. @gibson042 if you have nothing against, I'd merge this PR as-is and track the remaining changes via #5468. Let me know what you think.

**Comment by mgol** [comment]
Followup work extracted to #5468 per the last team meeting.

**Pull request merged**